actionmailer/lib/action_mailer/base.rb

```diff
@@ -888,9 +894,12 @@ def set_content_type(m, user_content_type, class_default) # :doc:
       # If it does not find a translation for the +subject+ under the specified scope it will default to a
       # humanized version of the <tt>action_name</tt>.
       # If the subject has interpolations, you can pass them through the +interpolations+ parameter.
+      # The locale may be passing in the <tt>locale</tt> key, otherwise the mailer class default or
+      # `I18n.locale` will be used.
       def default_i18n_subject(interpolations = {}) # :doc:
         mailer_scope = self.class.mailer_name.tr("/", ".")
-        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize))
+        locale = interpolations.key?(:locale) ? interpolations[:locale] : compute_default(self.class.default[:locale])
+        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize, locale: locale))
```

Comment:
If `interpolations.key?(:locale)` is true would not this method call have two `locale` options with the later overriding the former argument? I think we should only pass this option if `interpolations.key?(:locale)` is false.

Reply:
I'm not sure I understand your question. Is this a comment on style or functionality? 

The following is functionally equivalent. Do you prefer this approach?
```
        options = interpolations.key?(:locale) ? {} : { locale: compute_default(self.class.default[:locale]) }
        I18n.t(:subject, **interpolations.merge(scope: [mailer_scope, action_name], default: action_name.humanize), **options)
```

Reply:
@rafaelfranca Ping :-)

Reply:
@rafaelfranca Ping :-)